utils/compose.py

```diff
@@ -0,0 +1,7 @@
+import functools
+
+
+def compose(*functions):
+    def compose2(f, g):
+        return lambda x: f(g(x))
+    return functools.reduce(compose2, functions[::-1], lambda x: x)
```

Comment:
The gross part is that for some reason `functools.reduce` goes through `functions` backwards, and since `functions` is really a tuple, I can't use `functions.reverse()`. Thus the ugliness that is `functions[::-1]`. 🙁